esphome/__main__.py

```diff
@@ -106,6 +113,16 @@ def choose_upload_log_host(
     return choose_prompt(options, purpose=purpose)
 
 
+def mqtt_logging_enabled(mqtt_config):
+    if mqtt_config[CONF_LOG_TOPIC] is None:
+        return False
+    if CONF_TOPIC not in mqtt_config[CONF_LOG_TOPIC]:
+        return False
+    if mqtt_config[CONF_LOG_TOPIC].get(CONF_LEVEL, None) == "NONE":
+        return False
+    return True
```

Comment:
Always pull out a variable if it is going to be used multiple times
```suggestion
def mqtt_logging_enabled(mqtt_config):
	log_topic = mqtt_config[CONF_LOG_TOPIC]
    if log_topic is None:
        return False
    if CONF_TOPIC not in log_topic:
        return False
    if log_topic.get(CONF_LEVEL, None) == "NONE":
        return False
    return True
```